### PR TITLE
feature: configurable python (OpenAI) client

### PR DESCRIPTION
**Pull request by ibolmo**
All public interfaces that work with OpenAI transitively now have an optional AutoEvalCient client option. If not provided, we'll prepare and handle the OpenAI client as we do already.

At the moment you need to pass the client in each call, but in a follow-up commit you'll be able to set it globally.

### CODE REVIEW

File: py/autoevals/oai.py [hunk 1]
```diff
@@ -5,93 +5,146 @@
...
 from pathlib import Path
-from typing import Any
+from typing import Any, Optional
 
 PROXY_URL = "https://api.braintrust.dev/v1/proxy"
 
 
 @dataclass
-class OpenAIWrapper:
+class AutoEvalClient:
```
Comment (ankrgyl) [inline py/autoevals/oai.py:14]: maybe name it `LLMClient`? Or I guess it extends beyond LLMs theoretically?

**Review by ankrgyl — COMMENTED** [review]
lgtm

File: py/autoevals/oai.py (hunk 1 above)
Comment (ibolmo) [reply py/autoevals/oai.py:14]: yeah let's go with that 👍🏻

**Comment by ibolmo** [comment]
This will supersede #98

**Review by ankrgyl — APPROVED** [review]

**Pull request merged**